flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/VLCTextureView.java

```diff
@@ -0,0 +1,241 @@
+    private boolean debug = false;
+
+    private MediaPlayer mMediaPlayer = null;
+    private TextureRegistry.SurfaceTextureEntry mTextureEntry = null;
+    protected Context mContext;
+    private SurfaceTexture mSurfaceTexture = null;
+    private boolean wasPlaying = false;
+
+    private Handler mHandler;
+    private Runnable mLayoutChangeRunnable = null;
+
+    public VLCTextureView(final Context context) {
+        super(context);
+        mContext = context;
+        initVideoView();
+    }
+
+    public VLCTextureView(final Context context, final AttributeSet attrs) {
+        super(context, attrs);
+        mContext = context;
+        initVideoView();
+    }
+
+    public VLCTextureView(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        mContext = context;
+        initVideoView();
+    }
+
+    public void dispose() {
+        log("************************** dispose");
+
+        setSurfaceTextureListener(null);
+        removeOnLayoutChangeListener(this);
+
+        if (mLayoutChangeRunnable != null) {
+            mHandler.removeCallbacks(mLayoutChangeRunnable);
+            mLayoutChangeRunnable = null;
+        }
+
+        if (mSurfaceTexture != null) {
+            if (!mSurfaceTexture.isReleased()) {
+                mSurfaceTexture.release();
+            }
+            mSurfaceTexture = null;
+        }
+        mTextureEntry = null;
+        mMediaPlayer = null;
+        mContext = null;
+    }
+
+    private void initVideoView() {
+        mHandler = new Handler(Looper.getMainLooper());
+
+        setFocusable(false);
+        setSurfaceTextureListener(this);
+        addOnLayoutChangeListener(this);
+    }
+
+    public void setDebug(boolean value) {
+        debug = value;
+    }
+
+    public void setMediaPlayer(MediaPlayer mediaPlayer) {
+        log("************************** setMediaPlayer: " + mediaPlayer);
+
+        if (mediaPlayer == null) {
+            mMediaPlayer.getVLCVout().detachViews();
+        }
+
+        mMediaPlayer = mediaPlayer;
+
+        if (mMediaPlayer != null) {
+            mMediaPlayer.getVLCVout().attachViews(this);
+        }
+    }
+
+    public void setTextureEntry(TextureRegistry.SurfaceTextureEntry textureEntry) {
+        this.mTextureEntry = textureEntry;
+        this.updateSurfaceTexture();
+    }
+
+    private void updateSurfaceTexture() {
+        if (this.mTextureEntry != null) {
+            final SurfaceTexture texture = this.mTextureEntry.surfaceTexture();
+            if (!texture.isReleased() && (getSurfaceTexture() != texture)) {
+                setSurfaceTexture(texture);
+            }
+        }
+    }
+
+    @Override
+    public void onSurfaceTextureAvailable(SurfaceTexture surface, int width, int height) {
+        log("************************** onSurfaceTextureAvailable");
+
+        if (mSurfaceTexture == null || mSurfaceTexture.isReleased()) {

```

Comment:
Its better to close the second } if you are going to start with { 

its alot easier to read open and close brackets.